test/conftest.py

```diff
@@ -103,6 +109,34 @@ def git_urls():
             'port': None,
             'name': 'repo',
             'owner': 'owner',
+            'ref': None,
+        },
+        # gitlab and github allow omitting the .git
+        'user@example.com:owner/repo': {
+            'pathname': 'owner/repo',
+            'protocols': [],
+            'protocol': 'ssh',
+            'href': 'user@example.com:owner/repo',
+            'resource': 'example.com',
+            'user': 'user',
+            'port': None,
+            'name': 'repo',
+            'owner': 'owner',
+            'ref': None,
+        },
+        # npm and others allow specifying a branch or commit.  See
+        # https://docs.npmjs.com/files/package.json#git-urls-as-dependencies
```

Comment:
Why would this ever be used? `https://docs.npmjs.com/files/package.json#git-urls-as-dependencies`?  Thats an NPM url.

Reply:
That's just an example of one project that uses the convention.  yarn uses it, too.

I've been using it in my scripts to configure buildbot, and I find it useful.

Reply:
That said, it's not that much code to do without this feature.

Reply:
Sure, but npm isnt a git url neither is yarn urls.  I have not seen a git url having this format so I would have a hard time accepting a PR like this bc it's outside the scope of the project.

Reply:
I think it's a useful convention even outside of npm, but it's easy to implement as a tiny wrapper around git-url-parse, so closing.